presto-main/src/main/java/com/facebook/presto/type/ArrayParametricType.java

```diff
@@ -40,7 +40,9 @@ public String getName()
     @Override
     public Type createType(List<TypeParameter> parameters)
     {
-        checkArgument(parameters.size() == 1, "Array type expects exactly one type as a parameter, got %s", parameters);
+        if (parameters.isEmpty()) {
```

Comment:
Agreed. Yes, I've been thinking about changing and normalizing the nomenclature as part of the type system refactoring (see https://github.com/facebook/presto/wiki/Type-system). We'll probably do that after all the pending pull requests go in to avoid further conflicts.